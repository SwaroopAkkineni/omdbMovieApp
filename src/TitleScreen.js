import React, {Component} from 'react';
import {Dimensions, FlatList, Platform, Text, TouchableOpacity, View} from 'react-native';
import styles from '../StyleSheet';

export default class TitleScreen extends React.Component {
  render() {
    const { navigate } = this.props.navigation;
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>Welcome to React Native!</Text>
        <TouchableOpacity style={styles.todoItem}
                          onPress={() => navigate('TelevisionSeries', {showName: 'Parks and Recreation'})}>
          <Text style={styles.instructions}>Parks and Recreation</Text>
        </TouchableOpacity>
        <TouchableOpacity style={styles.todoItem}
                          onPress={() => navigate('TelevisionSeries', {showName: 'The Good Place'})}>
          <Text style={styles.instructions}>The Good PLace</Text>
        </TouchableOpacity>
        <TouchableOpacity style={styles.todoItem}
                          onPress={() => navigate('TelevisionSeries', {showName: 'The Office'})}>
          <Text style={styles.instructions}>The Office</Text>
        </TouchableOpacity>
      </View>
    );
  }
}
