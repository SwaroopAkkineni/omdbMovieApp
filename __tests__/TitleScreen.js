import React from 'react';
import 'react-native';
import TitleScreen from '../src/TitleScreen';

import renderer from 'react-test-renderer';
 const navigation = { navigate: jest.fn() };

test('renders correctly', () => {
  const tree = renderer.create(<TitleScreen navigation={navigation}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
