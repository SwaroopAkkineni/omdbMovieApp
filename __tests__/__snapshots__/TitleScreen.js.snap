// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#89D2DC",
      "flex": 1,
      "paddingLeft": 75,
      "paddingRight": 75,
      "textAlign": "left",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#101D42",
        "fontSize": 40.019999999999996,
        "fontWeight": "bold",
        "marginTop": 75,
      }
    }
  >
    Welcome to React Native!
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#89D2DC",
        "borderBottomWidth": 1.5,
        "borderColor": "#101D42",
        "flexDirection": "row",
        "height": 100.05,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#101D42",
        }
      }
    >
      Parks and Recreation
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#89D2DC",
        "borderBottomWidth": 1.5,
        "borderColor": "#101D42",
        "flexDirection": "row",
        "height": 100.05,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#101D42",
        }
      }
    >
      The Good PLace
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#89D2DC",
        "borderBottomWidth": 1.5,
        "borderColor": "#101D42",
        "flexDirection": "row",
        "height": 100.05,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#101D42",
        }
      }
    >
      The Office
    </Text>
  </View>
</View>
`;
